client/webserver/webserver.go

```diff
@@ -240,26 +241,34 @@ func (s *WebServer) Run(ctx context.Context) {
 		return
```

Comment:
I'm fairly sure that listeners are closed properly when calling srv.Shutdown, so I do not think it is necessary to close them separately. 
https://golang.org/pkg/net/http/#Server.Shutdown
> Shutdown works by first closing all open listeners...

Reply:
lol, ignore me, I just woke up

Reply:
That's my understanding.  Together with the graceful shutdown procedure of shutdown, that's why I replaced listener.Close with srv.Shutdown.  In some client hacking I noticed that `s.srv.Serve(listener)` would error with something about using a closed connection.  This seems to be the problem with closing the listeners manually and not using the Server's method.